### PR TITLE
Do runtime kfunc feature detection

### PR DESCRIPTION
**Pull request by danobi**
This helps decouple the build from the system bpftrace is built
on and eventually run on. So bpftrace built on older kernels/libraries
will still work correctly on newer kernels.

Output of --info now:
```
$ sudo ./build/src/bpftrace --info
System
  OS: Linux 5.5.15-200.fc31.x86_64 #1 SMP Thu Apr 2 19:16:17 UTC 2020
  Arch: x86_64

Build
  version: v0.10.0-7-g6054-dirty
  LLVM: 9
  foreach_sym: yes
  unsafe uprobe: no
  btf: yes
  bfd: yes
  bpf_attach_kfunc: yes

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no

Kernel features
  Instruction limit: 1000000
  Loop support: yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
````

### CODE REVIEW

**Comment by danobi** [comment]
On my arch box (whose kernel does not have kfunc or btf):
```
$ sudo ./src/bpftrace --info
System
  OS: Linux 5.6.3-arch1-1 #1 SMP PREEMPT Wed, 08 Apr 2020 07:47:16 +0000
  Arch: x86_64

Build
  version: v0.10.0-8-gac35
  LLVM: 9
  foreach_sym: yes
  unsafe uprobe: no
  btf: no
  bfd: yes
  bpf_attach_kfunc: yes

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: no
```

File: src/ast/semantic_analyser.cpp [hunk 1]
```diff
@@ -1882,12 +1882,15 @@ void SemanticAnalyser::visit(AttachPoint &ap)
...
-    bool supported = bpftrace_.btf_.has_data();
-#else
-    bool supported = false;
+#ifndef HAVE_BCC_KFUNC
+    error("kfunc/kretfunc not available for your linked against bcc version.",
+          ap.loc);
+    return;
 #endif
 
+    bool supported = (feature_.has_prog_kfunc() && bpftrace_.btf_.has_data())
```
Comment (olsajiri) [inline src/ast/semantic_analyser.cpp:1891]: you could just assign:

bool supported = feature_.has_prog_kfunc() && bpftrace_.btf_.has_data();

**Comment by olsajiri** [comment]
makes sense, apart from small comment it looks good to me

**Comment by fbs** [comment]
LGTM

File: src/ast/semantic_analyser.cpp (hunk 1 above)
Comment (danobi) [reply src/ast/semantic_analyser.cpp:1891]: oh whoops lol

**Pull request merged**